File: app/src/main/java/me/shreyasr/chatse/util/Logger.java
package me.shreyasr.chatse.util;

import android.util.Log;

/**
 * Utility class for logging.
 */
public class Logger {

    /**
     * Log an exception.
     *
     * @param source The class the exception came from.
     * @param message A message to give context on the exception.
     * @param e The exception (or throwable) itself.
     */
    public static void exception(Class source, String message, Throwable e) {
        Log.e(source.getSimpleName(), message, e);
    }

    /**
     * Log a message or event.
     *
     * @param source The class the message came from.
     * @param message The content of the message.
     */
    public static void message(Class source, String message) {
        Log.d(source.getSimpleName(), message);
    }
}

File: app/src/main/java/me/shreyasr/chatse/util/Logger.kt
package me.shreyasr.chatse.util

import android.util.Log

/**
 * Utility class for logging.
 */
object Logger {

    /**
     * Log an exception.

     * @param source  The class the exception came from.
     * *
     * @param message A message to give context on the exception.
     * *
     * @param e       The exception (or throwable) itself.
     */
    fun exception(source: Class<*>, message: String, e: Throwable) {
        Log.e(source.simpleName, message, e)
    }

    /**
     * Log a message or event.

     * @param source  The class the message came from.
     * *
     * @param message The content of the message.
     */
    fun message(source: Class<*>, message: String) {
        Log.d(source.simpleName, message)
    }

    fun event(source: Class<*>, message: String) {
        Log.i(source.simpleName, message)
    }

    fun debug(source: Class<*>, message: String) {
        Log.d(source.simpleName, message)
    }
}
